File: ebank/src/main/java/com/ebank/model/BankCard.java
package com.ebank.model;

import com.ebank.enums.Status;
import com.ebank.enums.CardType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import jakarta.persistence.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class BankCard {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String cardNumber;
    private String expirationDate;
    @Enumerated(EnumType.STRING)
    private CardType cardType;
    @Enumerated(EnumType.STRING)
    private Status status;
    private String blockRaison;

    @ManyToOne
    @JsonIgnore
    private Account account;
}

File: Ebank-Frontend/src/app/model/bank-card/bankcard.ts
import {Status} from "../../enum/status";
import {CardType} from "../../enum/card-type";


export class BankCard {
  id: number;
  cardNumber: string;
  expirationDate: string;
  cardType: CardType;
  status: Status;
  blockRaison: string;
  accountId: number;

  constructor(
    id: number,
    cardNumber: string,
    expirationDate: string,
    cardType: CardType,
    status: Status,
    blockRaison: string,
    accountId: number
  ) {
    this.id = id;
    this.cardNumber = cardNumber;
    this.expirationDate = expirationDate;
    this.cardType = cardType;
    this.status = status;
    this.blockRaison = blockRaison;
    this.accountId = accountId;
  }
}
